lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -63,6 +59,64 @@ struct ConnectInfo {
 using InnerRefMap =
     DenseMap<hw::InnerRefAttr, std::pair<hw::InnerSymAttr, StringAttr>>;
 
+//===----------------------------------------------------------------------===//
+// Naming Helpers
+//===----------------------------------------------------------------------===//
+
+static void appendName(StringRef name, SmallString<32> &output,
+                       bool toLower = false) {
+  if (name.empty())
+    return;
+  if (!output.empty())
+    output.append("_");
+  output.append(name);
+  if (!toLower)
+    return;
+  auto i = output.size() - name.size();
```

Comment:
What's this toLower business, by the way?

Reply:
(for instance naming, looks like?)

Reply:
Yes it's just for naming the instance of a layer-turned-module.